src/poslenstrings.jl

```diff
@@ -0,0 +1,526 @@
+primitive type PosLen 64 end
+PosLen(x::UInt64) = Core.bitcast(PosLen, x)
+UInt64(x::PosLen) = Core.bitcast(UInt64, x)
+
+Base.convert(::Type{PosLen}, x::UInt64) = PosLen(x)
+Base.convert(::Type{UInt64}, x::PosLen) = UInt64(x)
+
+const MISSING_BIT = 0x8000000000000000
+missingvalue(x) = (UInt64(x) & MISSING_BIT) == MISSING_BIT
+
+const ESCAPE_BIT = 0x4000000000000000
+escapedvalue(x) = (UInt64(x) & ESCAPE_BIT) == ESCAPE_BIT
+
+getpos(x) = (UInt64(x) & 0x3ffffffffff00000) >> 20
+getlen(x) = UInt64(x) & 0x00000000000fffff
+
+@noinline lentoolong(len) = throw(ArgumentError("len = $len too long"))
+
+@inline function PosLen(pos, len, missing=false, escaped=false)
+    len > 1048575 && lentoolong(len)
```

Comment:
OK. I see what happens :). Maybe add type annotations to `PosLen` arguments?

Reply:
type annotations? Do you mean code comments explaining things more?

Reply:
oh, I see what you mean. yeah, I can do that.